### PR TITLE
feat: update Facebook SDK to 5.x, fix iOS 13 login crash

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA**: https://jira.appcelerator.org/browse/MOD-2534

iOS and Android are working, docs have been updated. Ready for review!

### Native SDK builds:
- iOS: 5.5.0
- Android: 5.4.0

### CODE REVIEW

**Comment by hansemannn** [comment]
@vijaysingh-axway @janvennemann Can someone review this? We use it in production since ages.

**Comment by hansemannn** [comment]
@vijaysingh-axway Another ping to include this to 8.2.0 so FB apps continue to work

File: ios/Classes/Module/FacebookModule.m [hunk 1]
```diff
@@ -127,18 +130,22 @@ - (NSNumber *)accessTokenExpired
...
+  FBSDKAccessToken *newToken = [[FBSDKAccessToken alloc] initWithTokenString:[TiUtils stringValue:@"accessToken" properties:currentAccessToken]
+                                                              permissions:[currentAccessToken objectForKey:@"permissions"]
+                                                      declinedPermissions:[currentAccessToken objectForKey:@"declinedPermissions"]
+                                                       expiredPermissions:[currentAccessToken objectForKey:@"expiredPermissions"]
+                                                                    appID:[TiUtils stringValue:@"appID" properties:currentAccessToken]
+                                                                   userID:[TiUtils stringValue:@"userID" properties:currentAccessToken]
+                                                           expirationDate:[currentAccessToken objectForKey:@"exipirationDate"]
+                                                              refreshDate:[currentAccessToken objectForKey:@"refreshDate"]
+                                                 dataAccessExpirationDate:[currentAccessToken objectForKey:@"refreshDate"]];
```
Comment (vijaysingh-axway) [inline ios/Classes/Module/FacebookModule.m:147]: @hansemannn  dataAccessExpirationDate should be  @"refreshDate" or there should be new property for this?

**Comment by vijaysingh-axway** [comment]
@sgtcoolguy Build is failing due to "[ERROR] :  Unable to find any Xcode installations that supports iOS Simulator 13.0.". I guess need to use XCode  11 here as well.

**Review by vijaysingh-axway — COMMENTED** [review]
Looks good. Please  check a minor comment @hansemannn.

File: ios/Classes/Module/FacebookModule.m (hunk 1 above)
Comment (hansemannn) [reply ios/Classes/Module/FacebookModule.m:147]: I exposed it as "dataAccessExpirationDate" now.

**Comment by hansemannn** [comment]
@vijaysingh-axway The build issue is not module related and was fixed in ioslib 1.7.9+. I hope this can be merged

**Comment by sgtcoolguy** [comment]
I think it failed because it tried to build on an Xcode 11 beta 6 node with SDK 7.5.2.GA which doesn't work with Xcode 11. I'm updating the PR to try to build against 8.1.0.GA and see if that helps.

**Comment by sgtcoolguy** [comment]
Well we're getting closer. Now the iOS tests are hanging - and the karma runner doesn't really surface the logs as the app runs, so it's hard to tell what's going on. Maybe @janvennemann knows?

File: ios/Classes/Module/FacebookModule.m [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #import "TiBase.h"
 #import "TiHost.h"
 #import "TiUtils.h"
+#import "TiBlob.h"
```
Comment (hansemannn) [inline ios/Classes/Module/FacebookModule.m:19]: I'm curious why this does not cause issues for us right now. Maybe it works if the symbol is linked by other modules? Good catch in any case!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by sgtcoolguy** [comment]
I think ultimately our attempts to retain backwards compatibility were infeasible, so we'll go with this original PR where we bump the major version on iOS and break backwards compatibility - since Facebook has essentially killed off the messenger button.

One note: I think e should then remove the messenger button code from the example app inside the module here.

**Comment by keerthi1032** [comment]
FR Passed. Tested with Kitchensink apps Facebook and example app in facebook module.Login with facebook ,sharing and logout functionality works fine. 

Test Environment:
 Name                        = Mac OS X
  Version                     = 10.14.6
  Architecture                = 64bit
  Node.js
  Node.js Version             = 10.16.2
  npm Version                 = 6.9.0
Titanium CLI
  CLI Version                 = 5.2.1
Titanium SDK
  SDK Version                 = 8.2.0.v20190912104344
facebook module -7.0.0
Device -iPhone X iOS 11,iphone xr iOS 13.1
Simulator-iphone 11 iOS 13,iphone x iOS 12,iphone 6 iOS 11

**Pull request merged**